test_db.py

```diff
@@ -0,0 +1,231 @@
+        uuid.UUID(str(val))
+        return True
+    except ValueError:
+        return False
+
+
+# --------------------  USER FUNCTIONS --------------------
+class test_user_functions(unittest.TestCase):
+    def setUp(self):
+        self.con = db.connect_db()
+        self.cursor = self.con.cursor()
+        self.email = "test@email.gouv.ca"
+
+    def tearDown(self):
+        db.end_query(self.con, self.cursor)
+
+    def test_is_user_registered(self):
+        """
+        This test checks if the is_user_registered function returns the correct value
+        for a user that is not yet registered and one that is.
+        """
+
+        self.assertFalse(
+            user.is_user_registered(self.cursor, self.email),
+            "The user should not already be registered",
+        )
+
+        user_id = user.register_user(self.cursor, self.email)
+
+        self.assertTrue(is_valid_uuid(user_id), "The user_id is not a valid UUID")
+        self.con.rollback()
+
+    def test_get_user_id(self):
+        """
+        This test checks if the get_user_id function returns the correct UUID
+        """
+        user_id = user.register_user(self.cursor, self.email)
+        uuid = user.get_user_id(self.cursor, self.email)
+
+        self.assertTrue(is_valid_uuid(user_id), "The user_id is not a valid UUID")
+        self.assertTrue(is_valid_uuid(uuid), "The returned UUID is not a valid UUID")
+        self.assertEqual(
+            user_id, uuid, "The returned UUID is not the same as the registered UUID"
+        )
+
+        self.con.rollback()
+
+    def test_register_user(self):
+        """
+        This test checks if the register_user function returns a valid UUID
+        """
+        user_id = user.register_user(self.cursor, self.email)
+
+        self.assertTrue(is_valid_uuid(user_id), "The user_id is not a valid UUID")
+
+        self.con.rollback()
+
+
+# --------------------  SEED FUNCTIONS --------------------
+
+
+class test_seed_functions(unittest.TestCase):
+    def setUp(self):
+        self.con = db.connect_db()
+        self.cursor = self.con.cursor()
+        self.seed_name = "test"
+
+    def tearDown(self):
+        db.end_query(self.con, self.cursor)
+
+    def test_get_all_seeds_names(self):
+        """
+        This test checks if the get_all_seeds_names function returns a list of seeds
+        with at least the one seed we added
+        """
+        seed.new_seed(self.cursor, self.seed_name)
+        seeds = seed.get_all_seeds_names(self.cursor)
+
+        self.assertNotEqual(len(seeds), 0)
+        self.assertIn((self.seed_name,), seeds)
+
+        self.con.rollback()
+
+    def test_get_seed_id(self):
+        """
+        This test checks if the get_seed_id function returns the correct UUID
+        """
+        seed_uuid = seed.new_seed(self.cursor, self.seed_name)
+        fetch_id = seed.get_seed_id(self.cursor, self.seed_name)
+
+        self.assertEqual(seed_uuid, fetch_id)
+
+        self.con.rollback()
```

Comment:
`assertIsInstance()` :smile:

Reply:
The type will be a str so I use my built in function to verify the uuid
```suggestion
self.assertTrue(is_valid_uuid(fetch_id))
```